library/std/src/io/mod.rs

```diff
@@ -2502,6 +2505,21 @@ impl<T: BufRead> BufRead for Take<T> {
     }
 }
 
+impl<T> SizeHint for Take<T> {
+    #[inline]
+    fn lower_bound(&self) -> usize {
+        cmp::min(SizeHint::lower_bound(&self.inner) as u64, self.limit) as usize
```

Comment:
The cast to `usize` will overflow when `usize` is smaller than `u64`. Same in `upper_bound`, in the `Some`-arm, while the `None`-arm mitigates the overflow to `None`. In cases where `u64` is too big for `usize`, we probably want `usize::MAX`.

Reply:
Actually it cannot overflow because `SizeHint::lower_bound` returns an `usize`.
We know that `min(lower_bound() as u64, self.limit()) <= lower_bound()` so the result always fits in a `usize`, whatever the target's pointer width.

The same thing happens in `upper_bound`.